src/platform/packages/shared/kbn-server-route-repository/src/make_zod_validation_object.ts

```diff
@@ -19,6 +21,23 @@ export function makeZodValidationObject(params: ZodParamsObject) {
   };
 }
 
+export function makeZodResponsesValidationObject<
+  T extends TRouteResponse,
+  TReturnType = ExtractResponseStatusBodyTypes<T>
+>(responseSchema: T): RouteValidatorFullConfigResponse<TReturnType> {
+  const { unsafe, ...statusCodes } = responseSchema;
+  const response: RouteValidatorFullConfigResponse<TReturnType> = { unsafe };
+
+  for (const [statusCode, validation] of Object.entries(statusCodes)) {
+    response[parseInt(statusCode, 10)] = {
+      ...validation,
+      body: validation.body ? () => validation.body : validation.body,
```

Comment:
Just a drive-by comment: **response** validation is intended to run only when Kibana is in `dev` mode.

The way this works right now it looks like `validation.body` is eagerly instantiated then made "lazy" when passed to the router. It would be good to create this capability in a way that avoids instantiating response schemas in non-dev environments. [Docs for reference](https://docs.elastic.dev/kibana-dev-docs/genereating-oas-for-http-apis#2-route-definitions).

Reply:
@jloleysens the concern here is not to instantiate the runtime types (Zod in this case) in non-dev envs? The defined runtime schema for response types is only used for type checking and not the runtime response validation. 

If the response validation is always passed lazily to the core router, shouldn't it suffice or are there other concerns we've to take into account?

Reply:
> runtime schema for response types...always passed lazily to the core router

Not entirely sure I follow. Based on this code it looks like the response runtime schemas are not being lazily instantiated. They are `() => alreadyInstantiated` when to make them lazy they should be `() => z.object({...})`. This will avoid memory/heap consumption for things that should be dev only. Does that make sense?

Reply:
@jloleysens out of interest, do you have an idea of what the memory footprint is of that? asking because the Security Solution folks ship a ton of auto-generated zod schemas

Reply:
I have not looked at zod, but for config schema worst case scenario estimates upwards of 50MBs of idle/unused memory pressure.

Instantiating schemas/using memory is fine, but preferably only if they will be used!